### PR TITLE
Fix stale message index crash

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

[IOS-1727](https://linear.app/stream/issue/IOS-1727/swiftui-crash-in-chatchannelviewmodel)

### 🎯 Goal

Prevent stale message appearance callbacks from crashing `ChatChannelViewModel` when the displayed message snapshot is out of sync with the data source.

### 📝 Summary

- Guard message appearance handling against indexes that are not valid for the displayed `messages` snapshot.
- Keep pagination index checks aligned with the displayed message snapshot.
- Add a regression test for stale displayed messages vs data-source messages.

### 🛠 Implementation

`handleMessageAppear(index:scrollDirection:)` now validates the index against `messages.indices` before reading `messages[index]`, which matches the array used by `MessageListView` when calculating appearance indexes. The adjacent older/newer pagination checks now use the same displayed snapshot, and `checkForNewerMessages(index:)` has its own bounds guard because it also subscripts `messages`.

### 🎨 Showcase

N/A

### 🧪 Manual Testing Notes

- Open channels and threads

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [x] New code is covered by unit tests
- [ ] Documentation has been updated in the `docs-content` repo

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed a crash when loading or displaying more messages caused by an out-of-range message index.

* **Tests**
  * Added unit test coverage for message-display edge cases to prevent regressions.

* **Documentation**
  * Changelog updated with an “Upcoming” entry describing the fix.

[Review Change Stack](https://app.coderabbit.ai/change-stack/GetStream/stream-chat-swiftui/pull/1476?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by martinmitrevski — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # Upcoming
 
+### 🐞 Fixed
+- Fix a crash when stale message appearance callbacks use an index that is no longer valid for the displayed message list. [#1476](https://github.com/GetStream/stream-chat-swiftui/pull/1476)
```
Comment (nuno-vieira) [inline CHANGELOG.md]: It would be nice a shorter changelog
```suggestion
- Fix index out of range crash when loading more messages [#1476](https://github.com/GetStream/stream-chat-swiftui/pull/1476)
```

**Review by nuno-vieira — APPROVED** [review]
